src/query.cc

```diff
@@ -239,6 +243,8 @@ void DB::ApplyIndexUpdate(IndexUpdate *u) {
                  Use &use, int delta) {
     use.file_id =
         use.file_id == -1 ? u->file_id : lid2fid.find(use.file_id)->second;
+    if (usr == 0) return;
```

Comment:
When can `usr == 0` happen?

Reply:
When the data flow is from a non-trivial expression, which is considered a 'leaf' in the recursion.